pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -122,7 +131,26 @@ type PlayerSelector struct {
 	MaxAvailable int64 `json:"maxAvailable,omitempty"`
 }
 
-// ApplyDefaults applies default values to the PlayerSelector
+// CounterSelector is the filter options for a GameServer based on the count and/or available capacity.
+// Undefined or 0 for MaxCount or MaxAvailable means unlimited maximum.
+type CounterSelector struct {
+	MinCount     int64 `json:"minCount"`
+	MaxCount     int64 `json:"maxCount"`
+	MinAvailable int64 `json:"minAvailable"`
+	MaxAvailable int64 `json:"maxAvailable"`
+}
+
+// ListSelector is the filter options for a GameServer based on List available capacity and/or the
+// existence of a value in a List.
+// Undefined or 0 for MaxAvailable means unlimited maximum.
+// Undefined or "" for ValueInList means ignore field.
+type ListSelector struct {
+	MinAvailable int64  `json:"minAvailable"`
+	MaxAvailable int64  `json:"maxAvailable"`
+	ValueInList  string `json:"valueInList"`
```

Comment:
On the design we had `contains`, but I like how expressive this is. 🤔 WDYT of `containsValue` ? Since List is probably redundant since it's stored within a `lists` parent element anyway?

Reply:
Works for me